coalib/output/printers/LogPrinter.py

```diff
@@ -121,22 +124,23 @@ def __init__(self,
         self.logger = logging.getLogger()
 
         self._printer = printer
-        self.log_level = log_level
         self.timestamp_format = timestamp_format
+        if not printer:
+            self.log_level = log_level
 
     @property
     def log_level(self):
         """
         Returns current log_level used in logger.
         """
-        return self.logger.getEffectiveLevel()
+        return self.level
```

Comment:
Sorry for being unclear, this line is necessary. I was talking about the `if` statement a few lines above this.